data/lua/wml/random_placement.lua

```diff
@@ -14,7 +14,7 @@ wesnoth.wml_actions.random_placement = function(cfg)
 	local math_abs = math.abs	
 	local locs = wesnoth.get_locations(filter)
 	if type(num_items) == "string" then		
-		num_items = math.floor(loadstring("local size = " .. #locs .. "; return " .. num_items)())
+		num_items = math.floor(load("local size = " .. #locs .. "; return " .. num_items)())
```

Comment:
I didn't look too closely, so maybe.

I think it makes more sense to use a WFL formula for this. I understand that you can use global Lua functions if it's a Lua expression, but I wonder if that power is really needed. WFL has the benefit of forbidding side-effects, so it's impossible for the expression to unexpectedly change the game-state.